File: client/src/contracts/dtos/GetVacanciesDto.ts
export interface SalaryQuery {
    from?: number;
    to?: number;
    currency?: string;
}

export interface GetVacanciesDto {
    searchField?: string;
    experience?: string;
    employment?: string;
    schedule?: string;
    area?: string;
    salary?: SalaryQuery;
    minRating?: number;
    maxRating?: number;
    onlyWithReviews?: boolean;
    perPage?: number;
    pages?: number;
    page?: number;
}

File: src/Locator.Core/Infrastructure/HeadHunter/HeadHunter.Contracts/Dto/GetVacanciesDto.cs
namespace HeadHunter.Contracts.Dto;

public record GetVacanciesDto(
    string? SearchField,
    string? Experience,
    string? Employment,
    string? Schedule,
    string? Area,
    SalaryQuery? Salary,
    double? MinRating = null,
    double? MaxRating = null,
    bool? OnlyWithReviews = null,
    int? PerPage = null,
    int? Pages = null,
    int? Page = null);

public record SalaryQuery(string? Currency, int? From, int? To);
